### PR TITLE
Add compaction metrics and storeCompactionThrottlerCheckIntervalMs

### PR DESCRIPTION
**Pull request by zzmao**
Small PR to add compaction metrics and storeCompactionThrottlerCheckIntervalMs

### CODE REVIEW

File: ambry-api/src/main/java/com/github/ambry/config/StoreConfig.java [hunk 1]
```diff
@@ -116,6 +116,13 @@
   @Default("1*1024*1024")
   public final int storeCompactionOperationsBytesPerSec;
 
+  /**
+   * The check interval used in compaction rate throttler.
+   */
+  @Config("store.compaction.throttler.check.interval.ms")
+  @Default("-1")
```
Comment (jsjtzyy) [inline ambry-api/src/main/java/com/github/ambry/config/StoreConfig.java:123]: Please add some comment describing what does -1 mean. Looks like if it's <0,  rate is checked on every call.

File: ambry-store/src/main/java/com/github/ambry/store/BlobStore.java [hunk 2]
```diff
@@ -1212,7 +1212,7 @@ void compact(CompactionDetails details, byte[] bundleReadBuffer) throws IOExcept
     checkStarted();
     compactor.compact(details, bundleReadBuffer);
     checkCapacityAndUpdateReplicaStatusDelegate();
-    logger.trace("One cycle of compaction is completed on the store {}", storeId);
+    logger.info("One cycle of compaction is completed on the store {}", storeId);
```
Comment (jsjtzyy) [inline ambry-store/src/main/java/com/github/ambry/store/BlobStore.java:1215]: Maybe we can use `DEBUG` level?

File: ambry-store/src/main/java/com/github/ambry/store/BlobStoreCompactor.java [hunk 3]
```diff
@@ -551,17 +552,23 @@ private boolean isIndexSegmentUnderCopy(Offset indexSegmentStartOffset) {
   private boolean copyDataByLogSegment(LogSegment logSegmentToCopy, FileSpan duplicateSearchSpan)
       throws IOException, StoreException {
     logger.info("Copying data from {}", logSegmentToCopy);
+    long logSegmentStartTime = SystemTime.getInstance().milliseconds();
```
Comment (jsjtzyy) [inline ambry-store/src/main/java/com/github/ambry/store/BlobStoreCompactor.java:555]: can we use `time.milliseconds()` ?

**Review by jsjtzyy — APPROVED** [review]
lgtm after minor comments are addressed

File: ambry-store/src/main/java/com/github/ambry/store/BlobStoreCompactor.java (hunk 3 above)
Comment (justinlin-linkedin) [reply ambry-store/src/main/java/com/github/ambry/store/BlobStoreCompactor.java:555]: we should probably use System.currentMillis(); this is for metrics.

File: ambry-store/src/main/java/com/github/ambry/store/DiskManager.java [hunk 4]
```diff
@@ -494,7 +494,9 @@ boolean removeBlobStore(PartitionId id) {
   private Map<String, Throttler> getThrottlers(StoreConfig config, Time time) {
     Map<String, Throttler> throttlers = new HashMap<>();
     // compaction ops
-    Throttler compactionOpsThrottler = new Throttler(config.storeCompactionOperationsBytesPerSec, -1, true, time);
+    Throttler compactionOpsThrottler =
+        new Throttler(config.storeCompactionOperationsBytesPerSec, config.storeCompactionThrottlerCheckIntervalMs, true,
```
Comment (justinlin-linkedin) [inline ambry-store/src/main/java/com/github/ambry/store/DiskManager.java:498]: could you elaborate why adding a positive checkIntervalMs would change behavior of Throttler?

**Review by justinlin-linkedin — APPROVED** [review]

File: ambry-store/src/main/java/com/github/ambry/store/BlobStoreCompactor.java (hunk 3 above)
Comment (jsjtzyy) [reply ambry-store/src/main/java/com/github/ambry/store/BlobStoreCompactor.java:555]: The `time.miliseconds()` actually calls `System.currentMillis()` (`MockTime` isn't used in production code), so I think either is fine.

File: ambry-api/src/main/java/com/github/ambry/config/StoreConfig.java (hunk 1 above)
Comment (zzmao) [reply ambry-api/src/main/java/com/github/ambry/config/StoreConfig.java:123]: yes. You are right. Added in the comment.

File: ambry-store/src/main/java/com/github/ambry/store/BlobStore.java (hunk 2 above)
Comment (zzmao) [reply ambry-store/src/main/java/com/github/ambry/store/BlobStore.java:1215]: This is changed to `info` deliberately. When I look the log, there is log recording the start of compaction cycle, but no log for the end of the compaction cycle. 
In addition, this is not a repeated printed log. It should be ok to be in info level.

File: ambry-store/src/main/java/com/github/ambry/store/BlobStoreCompactor.java (hunk 3 above)
Comment (zzmao) [reply ambry-store/src/main/java/com/github/ambry/store/BlobStoreCompactor.java:555]: changed to time.

File: ambry-store/src/main/java/com/github/ambry/store/DiskManager.java (hunk 4 above)
Comment (zzmao) [reply ambry-store/src/main/java/com/github/ambry/store/DiskManager.java:498]: with -1, the rated is check every call. 
We also have a similar metric, `compactionCopyRateInBytes`, which shows stats per 60 seconds. 
Currently, if we set `storeCompactionOperationsBytesPerSec` to 5MB, the `compactionCopyRateInBytes` may go up to 15MB/s.
I suspect the difference come from the `-1`.

**Pull request merged**